### PR TITLE
New colour (invisible)

### PR DESCRIPTION
**Pull request by Kraots**
## Summary
Adds a new colour (invisible) to disnake.Colour which blends in with the embed's colour

## Checklist

- [ x] If code changes were made then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `black .`
- [ ] This PR fixes an issue
- [x ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by shiftinv — CHANGES_REQUESTED** [review]
The docstring should mention which theme this is for (light/dark/amoled), as it will only be invisible for one of them.

File: disnake/colour.py [hunk 1]
```diff
@@ -313,6 +313,15 @@ def dark_theme(cls: Type[CT]) -> CT:
         """
         return cls(0x36393F)
 
+    @classmethod
+    def invisible(cls: Type[CT]) -> CT:
+        """A factory method that returns a :class:`Colour` with a value of ``0x2f3136``.
+        This will blend in with the embed's colour.
+
+        .. versionadded:: 2.1
```
Comment (shiftinv) [inline disnake/colour.py:321]:
```suggestion
        .. versionadded:: 2.3
```

**Review by Kraots — COMMENTED** [review]
Fixed "versionadded" and added the theme "invisible" works on

**Review by Kraots — COMMENTED** [review]
Changed classmethod's name from "invisible" to "dark_invisible"

**Comment by abhigyantrips** [comment]
This isn't of much use if all theme users will not be having the same experience. This also isn't a truly "invisible" hex, which is why the name of the option can be misleading.

I believe such a PR was made for discord.py, but never merged for a similar reason (I will link the issue if I can find it). If collaborators believe this should be implemented, atleast have it as `invisible_light`, `invisible_dark`, and `invisible_amoled`.

**Comment by thesadru** [comment]
At that point, what about having a color for both blending with the background and blending with the embed color?

**Comment by abhigyantrips** [comment]
I guess we have to choose one? Either make it blend with the embed color or the background color. (From what I've implemented in my bots, I'd choose the former.)

**Comment by Kraots** [comment]
> This isn't of much use if all theme users will not be having the same experience. This also isn't a truly "invisible" hex, which is why the name of the option can be misleading.
> 
> I believe such a PR was made for discord.py, but never merged for a similar reason (I will link the issue if I can find it). If collaborators believe this should be implemented, atleast have it as `invisible_light`, `invisible_dark`, and `invisible_amoled`.

i changed the name to `dark_invisible` since the one right above it (`dark_theme`) is the one blending with discord's bg

**Comment by Kraots** [comment]
> At that point, what about having a color for both blending with the background and blending with the embed color?

the one blending with the bg is the one right above it, `dark_theme`

**Comment by abhigyantrips** [comment]
From the conversation we're having, this is starting to sound more like a gimmick than a viable feature. I believe the original embed colours were implemented with respect to a certain palette, so as to avoid giving beginners a hard time.

Since these colors are just a base palette, I'd rather that the developers wanting to implement the "invisible" embed color, get the hex directly from the theme.

**Comment by Kraots** [comment]
> From the conversation we're having, this is starting to sound more like a gimmick than a viable feature. I believe the original embed colours were implemented with respect to a certain palette, so as to avoid giving beginners a hard time.
> 
> Since these colors are just a base palette, I'd rather that the developers wanting to implement the "invisible" embed color, get the hex directly from the theme.

doesn't hurt to have it as a classmethod for disnake.Colour imo

**Comment by abhigyantrips** [comment]
Sure, it can be useful. But as I mentioned, since all users won't be experiencing it as "invisible", it's between *not* having them, or having an invisible colour for all three themes.

**Comment by shiftinv** [comment]
Just my two cents on this topic:
From what I can tell, the existing colors are either from the predefined role colors in the client, or from Discord's own [branding guidelines](https://discord.com/branding) (`greyple` was part of the older color scheme), with the exception of `dark_theme`.
If `dark_theme` wasn't already a thing, I'd vote to close this PR. However, given that it is a "unofficial" color that has made its way into dpy's/disnake's builtin colors, the idea of adding similar ones like embed colors of the different themes doesn't seem too bad.
*If* we plan to merge this, it should support the other themes as well, either through an additional parameter or using separate color `classmethod`s.

**Comment by abhigyantrips** [comment]
> If `dark_theme` wasn't already a thing, I'd vote to close this PR. However, given that it is a "unofficial" color that has made its way into dpy's/disnake's builtin colors, the idea of adding similar ones like embed colors of the different themes doesn't seem too bad.

Tbh I didn't even know `dark_theme` already existed haha. I agree that we'd have to support other colours as well, *if* we merge this. Though now I'm confused - isn't `invisible_dark` the same as `dark_theme`?

**Comment by shiftinv** [comment]
> isn't `invisible_dark` the same as `dark_theme`?

Almost, but not quite. One is the embed color, the other one is the background color: 
image

**Comment by abhigyantrips** [comment]
> Almost, but not quite. One is the embed color, the other one is the background color.

Ah, so I'd say we should only keep one of them (if we're keeping one at all). Otherwise the terminology used will also make things confusing for the developer.

**Comment by Kraots** [comment]
> > Almost, but not quite. One is the embed color, the other one is the background color.
> 
> Ah, so I'd say we should only keep one of them (if we're keeping one at all). Otherwise the terminology used will also make things confusing for the developer.

how is it making it confusing for developers? there's the docstring explaining it and the name itself is somewhat self-explanatory

**Comment by Kraots** [comment]
> > If `dark_theme` wasn't already a thing, I'd vote to close this PR. However, given that it is a "unofficial" color that has made its way into dpy's/disnake's builtin colors, the idea of adding similar ones like embed colors of the different themes doesn't seem too bad.
> 
> Tbh I didn't even know `dark_theme` already existed haha. I agree that we'd have to support other colours as well, _if_ we merge this. Though now I'm confused - isn't `invisible_dark` the same as `dark_theme`?

we could probably make the classmethod take a parameter named `theme` and have it default to `dark` or something like that :thinking:

**Comment by EQUENOS** [comment]
As shifinv mentioned, all current colors are either default role colors or from Discord's own branding guidelines. "Invisible" colors do not correspond to this category, plus no one really asked to add them.
In conclusion, I'd vote to close this PR.

**Comment by Kraots** [comment]
> As shifinv mentioned, all current colors are either default role colors or from Discord's own branding guidelines. "Invisible" colors do not correspond to this category, plus no one really asked to add them. In conclusion, I'd vote to close this PR.

alr :pensive:

**Pull request closed without merging**